go.mod

```diff
@@ -151,90 +151,90 @@ require (
 	github.com/vishvananda/netlink v1.2.1-beta.2
 	github.com/vmware/govmomi v0.42.0
 	go.uber.org/mock v0.4.0
-	golang.org/x/crypto v0.26.0
+	golang.org/x/crypto v0.27.0
 	golang.org/x/mod v0.20.0
-	golang.org/x/net v0.28.0
-	golang.org/x/oauth2 v0.22.0
+	golang.org/x/net v0.29.0
+	golang.org/x/oauth2 v0.23.0
 	golang.org/x/sync v0.8.0
-	golang.org/x/text v0.17.0
-	google.golang.org/api v0.196.0
-	google.golang.org/grpc v1.66.0
+	golang.org/x/text v0.18.0
+	google.golang.org/api v0.199.0
+	google.golang.org/grpc v1.67.0
 	gopkg.in/check.v1 v1.0.0-20201130134442-10cb98267c6c
 	gopkg.in/natefinch/lumberjack.v2 v2.2.1
 	gopkg.in/yaml.v2 v2.4.0
 	gopkg.in/yaml.v3 v3.0.1
 	helm.sh/helm/v3 v3.15.1
-	k8s.io/api v0.30.2
-	k8s.io/apiextensions-apiserver v0.30.1
-	k8s.io/apimachinery v0.30.2
-	k8s.io/apiserver v0.30.1
-	k8s.io/cli-runtime v0.30.1
+	k8s.io/api v0.31.1
+	k8s.io/apiextensions-apiserver v0.31.1
+	k8s.io/apimachinery v0.31.1
+	k8s.io/apiserver v0.31.1
+	k8s.io/cli-runtime v0.31.1
 	k8s.io/client-go v12.0.0+incompatible
 	k8s.io/helm v2.16.9+incompatible
-	k8s.io/kube-aggregator v0.30.1
-	k8s.io/kubectl v0.30.1
-	k8s.io/kubernetes v1.30.1
-	k8s.io/utils v0.0.0-20240502163921-fe8a2dddb1d0
+	k8s.io/kube-aggregator v0.31.1
+	k8s.io/kubectl v0.31.1
+	k8s.io/kubernetes v1.31.1
+	k8s.io/utils v0.0.0-20240711033017-18e509b52bc8
 	oras.land/oras-go/v2 v2.4.0
 	sigs.k8s.io/aws-iam-authenticator v0.6.17
-	sigs.k8s.io/cluster-api v1.7.3
-	sigs.k8s.io/controller-runtime v0.18.5
+	sigs.k8s.io/cluster-api v1.8.3
+	sigs.k8s.io/controller-runtime v0.19.0
```

Comment:
How are we deciding which version of `controller-runtime` to pick? `v1.8.3` capi uses `v1.18.5` https://github.com/kubernetes-sigs/cluster-api/blob/945c938ce3e093c71950e022de4253373f911ae8/go.mod#L51

Reply:
I tried that version but it was failing with 0.18.5 and in lasso it is also bumped with 0.19.0 hence i have choose 0.19.0

Reply:
Is it failing cuz lasso is at 0.19.0 or for some other reason?

Reply:
Lasso is at v0.19.0 because our PR bumped it to that version, but I now think`v0.18.5` would be more apt according to the upstream go.mod. I'm not sure what errors are causing the failure, but we can debug this on slack.

Reply:
controller-runtime 0.18 versions are using k8s 1.30 dependencies, because of which it is failing.

https://github.com/kubernetes-sigs/controller-runtime/blob/v0.18.5/go.mod

![image](https://github.com/user-attachments/assets/e3946344-0d61-46b3-a177-85852b676e4b)